storage-access.bs

```diff
@@ -163,6 +172,7 @@ To <dfn type="abstract-op">save the storage access flag set</dfn> for a [=partit
 partial interface Document {
   Promise&lt;boolean> hasStorageAccess();
   Promise&lt;undefined> requestStorageAccess();
+  Promise&lt;undefined> requestStorageAccessForSite(USVString site);
```

Comment:
Instead of `requestStorageAccessForSite(site)`, which introduces a new API and uses positional parameters, what if we added an options object to `requestStorageAccess`?

For example:

```js
await document.requestStorageAccess({
  site: "https://example.com"
})
```

This would help reduce potential confusion between `requestStorageAccess` and `requestStorageAccessForSite` (especially considering `requestStorageAccess` always autogrants in the top frame), and makes it so that future enhancements can be easily added without relying on positional parameters.

Reply:
Thanks for the feedback! I was viewing this as similar to `document.querySelector` vs `document.querySelectorAll`. A benefit I see of specifying a separate function is that it is easy to feature detect a new function, whereas it is less simple to detect whether that function would accept a parameter. There’s also the classic concern about getting property names right on a given call (i.e., `document.requestStorageAccess({site:'a.com'})` vs `document.requestStorageAccess({url:'a.com'})`), though this is probably minor. I think it might also be simpler to have separate documentation for the two call patterns; `requestStorageAccess` is already somewhat complicated in how it works, and this would be one more thing for developers to remember. I think not specifying a site doesn't really make sense at the top level anyway, which might help with your concern about mixing up the two calls. All this said, I don’t feel strongly. I'd definitely love to hear counterarguments or other opinions.